python/pyarrow/parquet/core.py

```diff
@@ -329,6 +330,9 @@ def __init__(self, source, *, metadata=None, common_metadata=None,
                  decryption_properties=None, thrift_string_size_limit=None,
                  thrift_container_size_limit=None, filesystem=None):
 
+        if pre_buffer and not is_threading_enabled():
+            pre_buffer = False
```

Comment:
I am not sure why we are doing this in many different places. Presumably, this could only be done somewhere at the bottom of the call stack.

Reply:
This one can indeed be removed, because it is passed to `ParquetReader.open()` from `_parquet.pyx`, where you added a similar check.

Reply:
And looking at the other occurences of the check in this file, I think all of those can be removed, because they either eventually pass the `pre_buffer` keyword to `ParquetFile` (which is handled in _parquet.pyx ParquetReader) or to the pyarrow.dataset ParquetFileFormat/ ParquetFragmentScanOptions, which you already handled in _dataset_parquet.pyx